Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -343,7 +343,9 @@ private extension ProductStore {
                 guard let product = products.first(where: { $0.sku == sku }) else {
                     return onCompletion(.failure(ProductLoadError.notFound))
                 }
-                onCompletion(.success(product))
+                self.upsertStoredProductsInBackground(readOnlyProducts: [product], siteID: siteID, onCompletion: {
```

Comment:
Another option here is to use the `replaceProductLocally` method, which is just a wrapper for the action above but using 1) A single product rather that a collection, and 2) `shouldDeleteExistingProducts` is always false. 

I didn't see a strong reason to use one or the other, as 1 may change soon, and 2 already defaults to false in the method signature, but happy to change it if you think it's more descriptive.

Reply:
I suspect `replaceProductLocally` is intended for edits which originate from the phone... so it may not be the best choice even though would work. `upsert` is fine. 

Why will 1 change soon? Is that because of a potential move to a fuzzy search, or because we would make an API request for multiple specific SKUs and get all of the results back in one response?

Reply:
> Why will 1 change soon? Is that because of a potential move to a fuzzy search, or because we would make an API request for multiple specific SKUs and get all of the results back in one response?

The later: If we go ahead with M2, part of it was to scan multiple SKUs so we can get all of the results back in one response.